packages/astro/src/core/errors/overlay.ts

```diff
@@ -445,6 +481,27 @@ class ErrorOverlay extends HTMLElement {
 		this.root = this.attachShadow({ mode: 'open' });
 		this.root.innerHTML = overlayTemplate;
 
+		// theme toggle logic
+		const themeToggleButton = this.root.querySelector<HTMLButtonElement>('#theme-toggle');
+		if (
+			localStorage.astroErrorOverlayTheme === 'dark' ||
+			(!('astroErrorOverlayTheme' in localStorage) &&
+				window.matchMedia('(prefers-color-scheme: dark)').matches)
+		) {
+			this?.classList.add('astro-dark');
+		} else {
+			this?.classList.remove('astro-dark');
+		}
+		themeToggleButton?.addEventListener('click', () => {
+			if (localStorage.astroErrorOverlayTheme === 'dark') {
+				this?.classList.remove('astro-dark');
+				localStorage.astroErrorOverlayTheme = 'light';
+			} else {
+				this?.classList.add('astro-dark');
+				localStorage.astroErrorOverlayTheme = 'dark';
+			}
```

Comment:
I recently learned from a community member about [`classList.toggle()`](https://developer.mozilla.org/en-US/docs/Web/API/DOMTokenList/toggle), which lets us simplify this:

```suggestion
			const isDark = localStorage.astroErrorOverlayTheme === 'dark';
			this?.classList.toggle('astro-dark', !isDark);
			localStorage.astroErrorOverlayTheme = isDark ? 'light' : 'dark';
```

Reply:
> Liking this enhancement!
> 
> Just popping in with some design guidance. The placement on the left side above the main title is very prominent in the hierarchy.
> 
> I mocked up a more standard toggle switch with it less prominent in the top right here:
> 
> https://www.figma.com/file/KM7dKti4BKAprnqhY8EHgI/Error-Messages?node-id=313%3A915&t=ptVVf9yWWcUszwVl-1
> 
> Let me know if I can provide any other guidance.

Here's the final result @doodlemarks 

![ezgif com-gif-maker (2)](https://user-images.githubusercontent.com/81974850/213046188-a8ea6fb4-ec40-4440-929d-b7f698aefc35.gif)